contrib/database/sql/option.go

```diff
@@ -47,3 +47,55 @@ func WithAnalyticsRate(rate float64) RegisterOption {
 		}
 	}
 }
+
+type openConfig struct {
+	serviceName   string
+	analyticsRate float64
+	dsn           string
+}
+
+// OpenOption represents an option that can be passed to Open or OpenDB.
+type OpenOption func(*openConfig)
+
+func openDefaults(cfg *openConfig) {
+	cfg.analyticsRate = math.NaN()
+}
+
+// OpenWithServiceName sets the given service name for the opened database.
+func OpenWithServiceName(name string) OpenOption {
+	return func(cfg *openConfig) {
+		cfg.serviceName = name
+	}
+}
+
+// OpenWithAnalytics enables Trace Analytics for all started spans.
+func OpenWithAnalytics(on bool) OpenOption {
+	return func(cfg *openConfig) {
+		if on {
+			cfg.analyticsRate = 1.0
+		} else {
+			cfg.analyticsRate = math.NaN()
+		}
+	}
+}
+
+// OpenWithAnalyticsRate sets the sampling rate for Trace Analytics events
+// correlated to started spans.
+func OpenWithAnalyticsRate(rate float64) OpenOption {
+	return func(cfg *openConfig) {
+		if rate >= 0.0 && rate <= 1.0 {
+			cfg.analyticsRate = rate
+		} else {
+			cfg.analyticsRate = math.NaN()
+		}
+	}
+}
+
+// OpenWithDataSourceName allows the data source name to be provided when
+// using OpenDB and a driver.Connector.
+// The value is used to automatically set tags on spans.
+func OpenWithDataSourceName(name string) OpenOption {
```

Comment:
Why not `WithDSN`? Do we need the long version?

Reply:
Updated this.